ejb3/src/main/java/org/jboss/as/ejb3/tx/CMTTxInterceptor.java

```diff
@@ -83,7 +83,7 @@ protected void endTransaction(final TransactionManager tm, final Transaction tx)
                 throw EjbLogger.ROOT_LOGGER.wrongTxOnThread(tx, tm.getTransaction());
             }
             final int txStatus = tx.getStatus();
-            if (txStatus == Status.STATUS_ACTIVE) {
+            if (txStatus == Status.STATUS_ACTIVE || txStatus == Status.STATUS_PREPARED || txStatus == Status.STATUS_PREPARING || txStatus == Status.STATUS_COMMITTING || txStatus == Status.STATUS_COMMITTED) {
```

Comment:
Prior to WFLY-1346, we used to call tm.commit() for everything but MARKED_ROLLBACK, so we have a lot of experience with handling PREPARED/PREPARING/COMMITTING/COMMITTED as you changed them to.